lib/factory_girl/linter_with_traits.rb

```diff
@@ -0,0 +1,37 @@
+module FactoryGirl
+  class LinterWithTraits < LinterBase
+
+    class FactoryTraitError < FactoryError
+      def initialize(wrapped_error, factory, trait_name)
+        super(wrapped_error, factory)
+        @trait_name = trait_name
+      end
+
+      def location
+        "#{@factory.name}+#{@trait_name}"
+      end
+    end
+
+    private
+
+    def calculate_invalid_factories
```

Comment:
Assignment Branch Condition size for calculate_invalid_factories is too high. [15.65/15]